File: src/test/java/removeduplicatesfromsortedarray/SolutionTest.java
package removeduplicatesfromsortedarray;

import org.assertj.core.api.Assertions;
import org.junit.jupiter.api.Test;

class SolutionTest {

    private final Solution solution = new Solution();

    @Test
    public void testOne() {
        int[] nums = {1, 1, 2};

        int uniqNumbers = solution.removeDuplicates(nums);

        Assertions.assertThat(uniqNumbers).isEqualTo(2);
    }

    @Test
    public void testTwo() {
        int[] nums = {0, 0, 1, 1, 1, 2, 2, 3, 3, 4};

        int uniqNumbers = solution.removeDuplicates(nums);

        Assertions.assertThat(uniqNumbers).isEqualTo(5);
    }

    @Test
    public void shouldPassWhenBothNumbersAreEqual() {
        int[] nums = {1, 1};

        int uniqNumbers = solution.removeDuplicates(nums);

        Assertions.assertThat(uniqNumbers).isEqualTo(1);
    }
}

File: src/test/kotlin/dev/sanket/removeduplicatesfromsortedarray/SolutionTest.kt
package dev.sanket.removeduplicatesfromsortedarray

import assertk.assertThat
import assertk.assertions.isEqualTo
import dev.sanket.dev.sanket.removeduplicatesfromsortedarray.Solution
import java.util.stream.Stream
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.Arguments
import org.junit.jupiter.params.provider.MethodSource

class SolutionTest {

    private val solution = Solution()

    @ParameterizedTest
    @MethodSource("testData")
    fun shouldPass(nums: IntArray, expected: Int) {
        val k = solution.removeDuplicates(nums)

        assertThat(k).isEqualTo(expected)
    }

    companion object {

        @JvmStatic
        fun testData(): Stream<Arguments> = Stream.of(
            Arguments.of(intArrayOf(1), 1),
            Arguments.of(intArrayOf(1, 1, 1, 2, 2, 3), 5),
            Arguments.of(intArrayOf(0, 0, 1, 1, 1, 1, 2, 3, 3), 7)
        )
    }
}